schachmat/wego
# enhanced error handling for open weather API errors

### Motivation and Context

This change was requested in this issue #176

### Description
I've handled error and success case that is coming from open weather API where I am checking the response code from the API and then printing it to the terminal, The marshalling was not happening because API sends Code as a string value for some of cases but if there is an invalid API key they send Code value is coming as an integer value I've handled this and just checking the type and marshalling it to a new struct called `openWeatherErrorResponse` 

### Steps for Testing
To test we can check different API responses like - 
1. Invalidate API key
2. Invalidate the open weather API
3. For success case everything should work like it used be.

### Screenshots

#### Invalid API key
![image](https://github.com/user-attachments/assets/c1f0d93d-54c0-4693-867b-31d3d706d37c)

#### Invalid API route
![image](https://github.com/user-attachments/assets/2ed9e23d-7994-4289-b834-e631036004fa)

#### Success
![image](https://github.com/user-attachments/assets/0e034c2b-3b3f-47e9-b281-733dbef8b615)
